opm/material/densead/FastSmallVector.hpp

```diff
@@ -0,0 +1,164 @@
+#include <array>
+#include <cstring>
+
+namespace Opm {
+namespace DenseAd {
+
+template <typename ValueType, unsigned N>
+class FastSmallVector
```

Comment:
I think that `FastSmallArray` would be a better class name because for me the term `Vector` is associated with adherence to the vector axioms. (yes, I think that `std::vector` is a horrible name for a dynamic array.)

I don't mind too much if you leave the class name as-is, though.